run.cc

```diff
@@ -51,29 +52,183 @@ bool testCopyConstructor() {
+  n = delta_y/sigma_delta_y;
+  cout << "delta_y AB: " << delta_y << endl;
+  cout << "sigma_delta_y AB: " << sigma_delta_y << endl;
+  cout << "n = delta_y/sigma_delta_y: " << n << endl; 
+  
   // here is the data from experiment A
   cout << "bin 27: from " << datA.binLow(27) << " to " << datA.binHigh(27)
        << endl;
   cout << "measurement of experiment A in bin 27: " << datA.measurement(27)
        << endl;
 
+  // Average two datasets (which is only done if they are compatible within 3 standard-deviations): 
+  Data avgAB = datA + datB;
+  Data avgAC = datA + datC;
+  Data avgAD = datA + datD;
+  Data avgBC = datB + datC;
+  Data avgBD = datB + datD;
+  Data avgCD = datC + datD;
+
+  // Check which experiments are compatible with each other within 2 standard-dev.: 
+  cout << "Number of differences between expA and expB with 2 standard-dev. " << meas[0].checkCompatibility(meas[1], 2) << endl ; 
+  cout << "Number of differences between expA and expC with 2 standard-dev. " << meas[0].checkCompatibility(meas[2], 2) << endl; 
+  cout << "Number of differences between expA and expD with 2 standard-dev. " << meas[0].checkCompatibility(meas[3], 2) << endl; 
+  cout << "Number of differences between expB and expC with 2 standard-dev. "  << meas[1].checkCompatibility(meas[2], 2) << endl; 
+  cout << "Number of differences between expB and expD with 2 standard-dev. "  <<meas[1].checkCompatibility(meas[3], 2) << endl; 
+  cout << "Number of differences between expC and expD with 2 standard-dev. "  << meas[2].checkCompatibility(meas[3], 2) << endl;
+
+  meas[0].returnOutliers(meas[1], 2);
+  meas[0].returnOutliers(meas[2], 2);
+  meas[0].returnOutliers(meas[3], 2);
+  meas[1].returnOutliers(meas[2], 2);
+  meas[1].returnOutliers(meas[3], 2);
+  meas[2].returnOutliers(meas[3], 2); 
+
+  cout << endl;
+  
+  // Check which experiments are compatible with each other within 3 standard-dev.: 
+  cout << "Number of differences between expA and expB with 3 standard-dev. " << meas[0].checkCompatibility(meas[1], 3) << endl ; 
+  cout << "Number of differences between expA and expC with 3 standard-dev. " << meas[0].checkCompatibility(meas[2], 3) << endl; 
+  cout << "Number of differences between expA and expD with 3 standard-dev. " << meas[0].checkCompatibility(meas[3], 3) << endl; 
+  cout << "Number of differences between expB and expC with 3 standard-dev. "  << meas[1].checkCompatibility(meas[2], 3) << endl; 
+  cout << "Number of differences between expB and expD with 3 standard-dev. "  <<meas[1].checkCompatibility(meas[3], 3) << endl; 
+  cout << "Number of differences between expC and expD with 3 standard-dev. "  << meas[2].checkCompatibility(meas[3], 3) << endl; 
+
+  meas[0].returnOutliers(meas[1], 3);
+  meas[0].returnOutliers(meas[2], 3);
+  meas[0].returnOutliers(meas[3], 3);
+  meas[1].returnOutliers(meas[2], 3);
+  meas[1].returnOutliers(meas[3], 3);
+  meas[2].returnOutliers(meas[3], 3);   
+
```

Comment:
In 2a) you didn't answer how many deviations you would typically expect even if the data were compatible. If you draw two samples from the same distribution, you would still expect some variation and even the probability to see a datapoint that has a deviation of 3 sigma is non-zero. So what you should have done here is just plug in the gaussian probability outside of 3 sigma (and 2 sigma) and multiply this with the number of datapoints, then see if the number of actual deviations agrees with what you computed.  Since your print out is correct, you'll still get 0.5 points for 2a)